src/formula.jl

```diff
@@ -213,3 +213,23 @@ function Formula(t::Terms)
     append!(rhs.args,t.terms)
     Formula(lhs,rhs)
 end
+
+"""
+    dropterm(f::Formula, trm::Symbol)
+
+Return a copy of `f` without the term `trm`.
+
+# Examples
+```jl
+julia> dropterm(foo ~ 1 + bar + baz, :bar)
+foo ~ 1 + baz
+```
+"""
+function dropterm(f::Formula, trm::Symbol)
+    rhs = copy(f.rhs)
+    args = rhs.args
+    if !(Meta.isexpr(rhs, :call) && args[1] == :+ && (tpos = findlast(args, trm)) > 0)
```

Comment:
I would like to discourage formulas like `y ~ a` and require either `y ~ 1 + a` or `y ~ 0 + a`

Also, what should `dropterm(y ~ a, :a)` return?  The lhs of a formula can be `nothing` but I don't know how to make sense of a rhs of `nothing`.